accepted/future-releases/unified-collections/feature-specification.md

```diff
@@ -0,0 +1,887 @@
+
+            1.  Wait for `f` to complete. If `f` completes with an error
+                `error2` and stack trace `stack2`, then complete `streamDone`
+                with error `error2` and stack trace `stack2`. Otherwise complete
+                `streamDone` with error `error` and stack trace `stack`.
+
+        *   On a done event, complete `streamDone` with the value `null`.
+
+    1.  Evaluation of the `for-in` element completes when `streamDone`
+        completes. If `streamDone` completes with an error `error` and stack
+        trace `stack`, then evaluation of the element throws `error` with stack
+        trace `stack`, otherwise the evaluation completes successfully.
+
+1.  Else, `element` is a C-style `for` element:
+
+    1.  Evaluate the initializer clause of the element, if there is one.
+
+    1.  Loop:
+
+        1.  If the initializer clause declares a variable, create a new
+            namespace and bind that variable in that namespace.
+
+        1.  Evaluate the condition expression to a value `condition`. If there
+            is no condition expression, use `true`.
+
+        1.  If the Boolean conversion of `condition` is not `true`, exit the
+            loop.
+
+        1.  Evaluate the body element using this procedure in the namespace of
+            the variable declared by the initializer clause if there is one.
+
+        1.  If there is an increment clause, execute it.
+
+The procedure theoretically supports a mixture of expressions and map entries,
+but the static semantics prohibit an actual literal containing that. Once the
+*result* series of values or entries is produced from the tree of elements:
+
+### Lists
+
+1.  The result of the literal expression is a fresh instance of a class that
+    implements `List<E>` containing the values in *result*, in order. If the
+    literal is constant, the list is canonicalized and immutable, otherwise it
+    is not.
+
+### Sets
+
+1.  Create a fresh instance *set* of a class that implements `Set<E>`.
+
+1.  For each *value* in *result*:
+
+    1.  If *set* does not contain any value *existing* that is equal to *value*
+        according to *existing*'s `==` operator, then add *value* to *set*.
+
+    1.  Else, if the literal is constant, it is a compile-time error.
+
+    1.  Else, do nothing. *Duplicates are discarded and the first one wins.*
+
+1.  The result of the literal expression is *set*. If the literal is constant,
+    canonicalize it make the set immutable.
+
+### Maps
+
+1.  Allocate a fresh instance *map* of a class that implements `LinkedHashMap<K,
+    V>`.
+
+1.  For each entry *key*: *value* in *result*:
+
+    1.  If *map* does not contain any key *existing* that is equal to *key*
+        according to *existing*'s `==` operator, then insert *key*: *value* into
+        *map*.
+
+    1.  Else, if the literal is constant, it is a compile-time error.
+
+    1.  Else, do nothing. *Duplicate keys are discarded and the first entry
```

Comment:
Actually, you need to update the value, but not change the key's position in the iteration (just like `[]=` on `LinkedHashMap`, uncoincidentally).